pantheon-bundle/frontend/src/app/searchFilter.tsx

```diff
@@ -1,36 +1,381 @@
+    const versionUUID = this.state.allProducts[product].filter((e) => e.label === id)[0][uuidKey]
+    const productUUID = this.state.productsUUID[product]
+    if (versionUUID === "All") {
+      let prodQuery = this.state.productsQueryParam
+      prodQuery = prodQuery.replace("product=" + productUUID, '')
+      if (prodQuery === '&') {
+        prodQuery = ''
+      }
+      if (prodQuery.includes("&&")) {
+        prodQuery = prodQuery.replace('&&', '&')
+      }
+      this.setState({ chipGroups: copyOfChipGroups, productsQueryParam: prodQuery }, () => {
+        this.setQuery();
+      });
+    } else {
+      let verQuery = this.state.productversionsQueryParam
+      verQuery = verQuery.replace("productversion=" + versionUUID, '')
+      if (verQuery === '&') {
+        verQuery = ''
+      }
+      if (verQuery.includes("&&")) {
+        verQuery = verQuery.replace('&&', '&')
+      }
+      this.setState({ chipGroups: copyOfChipGroups, productversionsQueryParam: verQuery }, () => {
+        this.setQuery();
+      });
+    }
+  };
+
+  private addChipItem = () => {
+    const copyOfChipGroups = this.state.chipGroups;
+    let exist = false
+    let index = 0
+    for (let i = 0; copyOfChipGroups.length > i; i++) {
+      const category = copyOfChipGroups[i].category
+      if (category === this.state.productValue) {
+        exist = true
+        index = i
+        break
+      }
+    }
+    if (exist) {
+      let chipExists = false
+      for (const i of copyOfChipGroups[index].chips) {
+        const chip = i
+        if (chip === this.state.versionSelected) {
+          chipExists = true
+        }
+
+      }
+      if (!chipExists && this.state.versionSelected !== "Select a Version") {
+        copyOfChipGroups[index].chips.push(this.state.versionSelected);
+      }
+    } else {
+      copyOfChipGroups.push({
+        category: this.state.productValue,
+        chips: [this.state.versionSelected]
+      })
+    }
+
+    const uuidKey = "value"
+    const versionUUID = this.state.allProducts[this.state.productValue].filter((e) => e.label === this.state.versionValue)[0][uuidKey]
+    // If version is All just add the product.
+    let prodQuery = this.state.productsQueryParam
+    let verQuery = this.state.productversionsQueryParam
+    if (versionUUID === "All") {
+      if (this.state.productsQueryParam !== '') {
+        prodQuery += '&'
+      }
+      prodQuery += "product=" + this.state.productsUUID[this.state.productValue]
+    } else if (versionUUID !== "") {
+      if (this.state.productversionsQueryParam !== '') {
+        verQuery += '&'
+      }
+      verQuery += "productversion=" + versionUUID
+    }
+
+    this.setState({ chipGroups: copyOfChipGroups, productsQueryParam: prodQuery, productversionsQueryParam: verQuery }, () => {
+      this.setQuery();
+    });
+  };
+
+  // Should be called after each change of state
+  private setQuery = () => {
+    let searchQuery = ""
+    if (this.state.searchText !== "") {
```

Comment:
I suggest we use (mystring.trim() !== "") to evaluate if mystring is empty or not. trim removes whitespace and newlines